### PR TITLE
Remove non rpc interface methods `AnalogNames` and `DigitalInterruptNames` from board

### PR DESCRIPTION
**Pull request by randhid**
We can now do this cleanup chore since `Status` and `StatusStream` are completely unused and the reason for these rpcs to exist is now gone.

Tested with Viam server running on a mac with a few fake modular boards.

Will probably wait to merge until SDK work is complete.

### CODE REVIEW

**Comment by martha-johnston** [comment]
can modular boards still optionally implement these functions? or will it break them all, including pi?

**Review by JohnN193 — APPROVED** [review]

**Comment by JohnN193** [comment]
> can modular boards still optionally implement these functions? or will it break them all, including pi?

modules should be unaffected by this. They would just have extra exported methods on their structs

File: components/board/client.go [hunk 1]
```diff
@@ -92,22 +92,6 @@ func (c *client) GPIOPinByName(name string) (GPIOPin, error) {
 	}, nil
 }
 
-func (c *client) AnalogNames() []string {
-	if len(c.info.analogNames) == 0 {
```
[the remark this answers is not in the document]
Comment (randhid) [reply components/board/client.go:96]: Will check it out.

**Comment by randhid** [comment]
> can modular boards still optionally implement these functions? or will it break them all, including pi?

> modules should be unaffected by this. They would just have extra exported methods on their structs

This is in fact what happens with an interface removal in go and it is harmless for board modules. With this change I'm no longer forcing module authors to implement these methods that could not be accessed.

You can consider the addition of an API to be a forward breaking change for modules. The module will need to implement the new method. Removing a method does not break a module, as the `AnalogNames` method can just sit there doing nothing in the module code after rdk is updated in that module. It's actually the opposite of the normal breaking API change rules. Good question to ask martha.

**Review by martha-johnston — APPROVED** [review]

**Comment by randhid** [comment]
Going to wait till I'm back from vacation to merge 

@martha-johnston good remote work is doing the corresponding changes in sdks - do go ahead and make tickets for this.

**Pull request merged**